web/packages/teleport/src/components/ClientSearcheableTableWithQueryParamSupport/ClientSearcheableTableWithQueryParamSupport.tsx

```diff
@@ -0,0 +1,53 @@
+/**
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { useHistory } from 'react-router';
+
+import { TableProps } from 'design/DataTable/types';
+import Table from 'design/DataTable';
+
+import {
+  decodeUrlQueryParam,
+  encodeUrlQueryParams,
+} from '../hooks/useUrlFiltering';
+
+export function ClientSearcheableTableWithQueryParamSupport<T>(
```

Comment:
Is there a client-searchable table without query param support? If not then I think we might as well drop `WithQueryParamSupport` from the name. ;)

If someone wants to have a client-searchable table that doesn't update the query params, then they can just use a plain `<DataTable>`

Reply:
BTW, does it mean that client-side search is always enabled unless I specify `serversideProps`?

Reply:
> Is there a client-searchable table without query param support?

yes, the query param support is opt in (you have to supply the `tableProps.clientSearch`) to make it work. i figured we can convert the rest of table by "need" basis

> BTW, does it mean that client-side search is always enabled unless I specify serversideProps?

if you use this table yes. if you supply `serversideProps`, the `clientSearch` gets ignored. the table that uses `serversideProps` provides us their own `serversideSearchPanel` component to handle its own server side searching.

I updated this tables props to disallow `serverSideProps`